test/windows/WsladiagTests.cpp

```diff
@@ -0,0 +1,143 @@
+/*++
+
+Copyright (c) Microsoft. All rights reserved.
+
+Module Name:
+
+    WsladiagTests.cpp
+
+Abstract:
+
+    This file contains smoke tests for wsladiag.
+
+--*/
+
+#include "precomp.h"
+#include "Common.h"
+#include <format>
+
+namespace WsladiagTests {
+class WsladiagTests
+{
+    WSL_TEST_CLASS(WsladiagTests)
+
+    TEST_CLASS_SETUP(TestClassSetup)
+    {
+        VERIFY_ARE_EQUAL(LxsstuInitialize(FALSE), TRUE);
+        return true;
+    }
+
+    TEST_CLASS_CLEANUP(TestClassCleanup)
+    {
+        LxsstuUninitialize(FALSE);
+        return true;
+    }
+
+    static std::wstring BuildWsladiagCmd(const std::wstring& args)
+    {
+        const auto exePath = wsl::windows::common::wslutil::GetBasePath() / L"wsladiag.exe";
```

Comment:
`GetBasePath()` won't work here since it will look up the current process's executable folder. 

To get the installation folder where wsladiag.exe is, use : `wsl::windows::common::wslutil::GetMsiPackagePath()`